pkg/app/build_azure.go

```diff
@@ -49,7 +96,21 @@ func azureCredentials(config Config) (*azure.Credentials, error) {
 		return nil, fmt.Errorf("failed to get tenant id: %w", err)
 	}
 
-	credentials, err := azure.NewCredentials(clientID, clientSecret, tenantID)
+	endpoint, err := config.GetWithError(PackerAzureCloudEndpointPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cloud endpoint: %w", err)
+	}
+	var cloudConfig cloud.Configuration
+	switch endpoint {
+	case string(AzureCloudEndpointChina):
+		cloudConfig = cloud.AzureChina
+	case string(AzureCloudEndpointGovernment):
+		cloudConfig = cloud.AzureGovernment
+
+	default:
+		cloudConfig = cloud.AzurePublic
+	}
```

Comment:
the global variables for cloudconfig does not set the endpoint (https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.1.2/cloud#pkg-variables)
I think you have to set it explicitly here. 
```go
// for AzureChina
cloudConfig.Services[cloud.ResourceManager] = cloud.ServiceConfiguration{
		Audience: "https://management.core.chinacloudapi.cn",
		Endpoint: "https://management.chinacloudapi.cn",
	}

// for AzureGovernment
	cloudconfig.Services[cloud.ResourceManager] = cloud.ServiceConfiguration{
		Audience: "https://management.core.usgovcloudapi.net",
		Endpoint: "https://management.usgovcloudapi.net",
	}
// for AzurePublic
	cloudconfig.Services[cloud.ResourceManager] = cloud.ServiceConfiguration{
		Audience: "https://management.core.windows.net/",
		Endpoint: "https://management.azure.com",
	}
```
ex: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/arm.go#L12

Reply:
the line you linked to is an init function that does that sets the [cloud.ResourceManager] field  in the Service for me with the right Audience and Endpoint 
 
When I spew the cloudConfig we see its configured correctly
 ```
 2022/08/24 12:57:26 writing new packer configuration to work/centos-7-1661371046-VMmAq
(cloud.Configuration) {
 ActiveDirectoryAuthorityHost: (string) (len=34) "https://login.microsoftonline.com/",
 Services: (map[cloud.ServiceName]cloud.ServiceConfiguration) (len=1) {
  (cloud.ServiceName) (len=15) "resourceManager": (cloud.ServiceConfiguration) {
   Audience: (string) (len=36) "https://management.core.windows.net/",
   Endpoint: (string) (len=28) "https://management.azure.com"
  }
 }
}
```